File: 55-jump-game/jump-game.cpp
class Solution {
public:
    bool canJump(vector<int>& nums) {
        int maxReach = 0;
        bool possible = false;
        for (int i=0;i<nums.size();i++){
            if(i>maxReach){return false;}
            else{
                if(maxReach<i+nums[i]){
                    maxReach = i+nums[i];
                } 
            }
            if (maxReach>=i){possible=true;}
            
        }
        return possible;
        
    }

               
};

File: 55-jump-game/jump-game.py
class Solution:
    def canJump(self, nums: List[int]) -> bool:
        maxReach = 0
        possible = False
        while possible == False:
            for i in range(0,len(nums)):
                if i > maxReach:
                    return False
                else: 
                    maxReach = max(maxReach, i + nums[i])
                    if maxReach>=i: possible = True
        return possible




        